src/ameba/config.cr

```diff
@@ -239,6 +239,13 @@ class Ameba::Config
     end
   end
 
+  private def default_globs
+    DEFAULT_GLOBS.select do |glob|
+      !Dir[glob].empty? ||
+        (glob.starts_with?("!") && !Dir["#{glob[1..-1]}/**/*.cr"].empty?)
+    end
+  end
```

Comment:
I would probably remove `!lib` part from the DEFAULT_GLOBS and include it conditionally here (as an opposite to excluding it). Also, DEFAULT_GLOBS would probably have to be renamed:

```crystal
MATCHING_SOURCES_GLOB = "**/*.cr"
```

WDYT?

Reply:
Thank you! I've pushed eb60b25 using a similar approach to what you described.

Let me know if there's anything else we can improve me merging this :)